Documentation/devel/architecture.md

```diff
@@ -102,3 +102,60 @@ This will activate all the reaper services when one of the targets is activated,
+* Store: in the store phase the fetched images will be saved to the local store. The local store is a cache for fetched images and additional related data.
+* Render: in the render phase, a renderer will pull the required images from the store and render them so they can be easily used as a stage2 content.
+
+
+These three logical blocks are implemented inside rkt in this way:
+
+```
+ ------------            ---------------                -------------                    ------------------------
+|            |          |               |              |             |     overlayfs    |                        |
+|  Fetchers  |--------->|  Image Store  |<-------------|  TreeStore  |<-----------------|  Stage1-2 fs contents  |
+|            |          |               |<----         |             |           -------|                        |
+ ------------            ---------------      \         -------------           /        ------------------------
+                                               \                               /
+                                                \       -----------------     /
+                                                 \     |                 |   /
+                                                  -----| Direct stage1-2 |---
+                                                       |   renderer      |
+                                                       |                 |
+                                                        -----------------
+```
+
+Currently rkt implements the appc spec, in future additional specs like (OCI) may be implemented keeping the same logic but changing the implementations.
+
+* Fetchers: the fetchers, depending on the provided image argument will do image discovery (if an image string is provided) and/or fetching (using the discovered urls or the provided urls). The fetchers may read some information from the store to check if an image is already in the store or to retrieve additional data like http caching information. Once fetched the images will be verified against their signatures and then saved in the Image Store. If an image has some dependencies, they will be discovered and fetched. See also [image fetching behavior](../image-fetching-behavior.md).
+* Image Store: the Image Store is used to store images (currently ACIs) and their related information.
+* The render phase can be done in different ways:
+ * Directly render the stage1-2 contents inside a pod. This will require more disk space and more stage1-2 preparation time.
+ * Render in the treeStore. The treestore is a cache of rendered images (currently ACIs). rkt, when using the treestore, instead of directly rendering the stage1-2 contents inside a pod everytime, will mount an overlayfs using the treestore rendered image as its lower dir.
```

Comment:
Second sentence:
"When using the treestore, rkt mounts an overlayfs with the treestore-rendered image as its lower directory."